### ISSUE TITLE
Undici error on Windows with https

### ISSUE THREAD

**Issue by ibenrh** [Azure/azure-functions-nodejs-library#235]
When running `func start` with `--useHttps` on Windows you get the error message:

```
Exception: Headers.append: ":method" is an invalid header name.
Stack: TypeError: Headers.append: ":method" is an invalid header name.
```

This is caused by any `undici` versions higher than `5.27.2`.

Example at: https://github.com/pellebjerkestrand/azure-functions-core-tools-invalid-header-method/tree/main

**Comment by ejizba** [comment]
Looks like we're not the [only](https://github.com/sveltejs/kit/issues/11365) [ones](https://github.com/nodejs/undici/issues/2415) hitting an error like this. It has to do with pseudo headers in http/2 and I'm not entirely sure what the recommended behavior is.

I can repro this myself on Windows with `--useHttps`, but not mac. I checked telemetry and I'm not seeing evidence of this error in Azure - seems like it's only a problem locally.

**Comment by steffennilsen** [comment]
I am running into the same issue with local hosting on Linux after enabling `--useHttps`, @ibenrh solution of downgrading `undici` solved the issue for me

**Comment by ibenrh** [comment]
Yeah, we've temporarily solved it that way too for now, @steffennilsen :)

**Comment by timtucker-dte** [comment]
Would starting up the local server with http2 support disabled fix this by preventing the browser from sending http2 pseudo headers?

**Comment by timtucker-dte** [comment]
Anyone else tested this recently?

Tried again with 4.6.1 / undici 5.28.5 today and https seems to be working.

**Comment by swapnil-nagar** [comment]
We have deprecated the undici dependency for @azure/functions package.
